src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,53 @@
+.calendar {
+  $days-of-the-week:
+    "mon" 6,
+    "tue" 5,
+    "wed" 4,
+    "thu" 3,
+    "fri" 2,
+    "sat" 1,
+    "sun" 0;
+
+  @each $day-of-the-week, $number-of-the-week in $days-of-the-week {
+    &__start-day--#{$day-of-the-week} .calendar__item:nth-child(1) {
+      margin-left: $number-of-the-week * ($cell-size + $date-gap);
+    }
+  }
+
+  @for $date-n from 28 through 31 {
+    &--calendar-length--#{$date-n} .calendar__item:nth-child(n+#{$date-n + 1}) {
+      display: none;
+    }
+  }
+
+  &__list {
+    display: flex;
+    flex-wrap: wrap;
+    gap: $date-gap;
+  }
+
+  &__item {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    @include square($cell-size);
+    background-color: $date-color;
+    border: 1px solid #000;
+    border-radius: 5px;
+    transition: all 0.5s;
+    cursor: pointer;
+
+    &:hover {
+      background-color: $date-color-hover;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $date from 1 through 31 {
```

Comment:
Consider using a more descriptive variable name for `$date` to improve code readability, such as `$date_number`.